storage_service/locations/models/archipelago.py

```diff
@@ -0,0 +1,264 @@
+
+import requests
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from lxml import etree
+
+from .location import Location
+
+# Core Django, alphabetical
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Archipelago(models.Model):
+    """Integration with Archipelago using the REST API."""
+
+    space = models.OneToOneField("Space", to_field="uuid", on_delete=models.CASCADE)
+
+    archipelago_url = models.URLField(
+        max_length=256,
+        verbose_name=_("Archipelago URL"),
+        help_text=_("Archipelago URL"),
+    )
+
+    archipelago_user = models.CharField(
+        max_length=64,
+        verbose_name=_("Archipelago username"),
+        help_text=_("Archiipelago username for authentication"),
+    )
+
+    archipelago_password = models.CharField(
+        max_length=64,
+        verbose_name=_("Archipelago password"),
+        help_text=_("Archipelago password for authentication"),
+    )
+
+    class Meta:
+        verbose_name = _("Archipelago via REST API")
+        app_label = "locations"
+
+    ALLOWED_LOCATION_PURPOSE = [Location.AIP_STORAGE]
+
+    def _upload_file(self, filename, source_path):
+        """Uploads zip file to Archipelago before creating new entity
+        so if upload fails, new entity not created"""
+        url = self.archipelago_url + "/jsonapi/node/aip/field_file_drop"
+        headers = {
+            "Content-Type": "application/octet-stream",
+            "Content-Disposition": f'file; filename="{filename}"',
+        }
+        try:
+            with open(source_path, "rb") as file:
+                response = requests.post(
+                    url,
+                    data=file,
+                    headers=headers,
+                    auth=(self.archipelago_user, self.archipelago_password),
+                )
+                response.raise_for_status()
+
+            if response.status_code == 201:
+                LOGGER.info("AIP file uploaded successfully!")
+                response_json = response.json()
+                fid = (
+                    response_json.get("data", {})
+                    .get("attributes", {})
+                    .get("drupal_internal__fid")
+                )
+                return fid
+            else:
+                LOGGER.error(
+                    f"File upload failed with status code {response.status_code}: {response.text}"
+                )
+        except (OSError, requests.exceptions.RequestException) as e:
+            LOGGER.error("Error during AIP upload to archipelago %s", str(e))
+
+    def extract_title_from_mets_xml(self, xml_string):
+        """Retrieves title from METs file or creates title from file name"""
+        try:
+            root = etree.fromstring(xml_string)
+            namespaces = {
+                "mets": "http://www.loc.gov/METS/",
+                "dc": "http://purl.org/dc/elements/1.1/",
+                "dcterms": "http://purl.org/dc/terms/",
+            }
```

Comment:
Namespaces are defined [here](https://github.com/UoEMainLibrary/archivematica-storage-service/blob/drupalFeature/storage_service/common/utils.py#L25-L37). I would suggest you could have import it here.